plugins/modules/remote_management/lenovoxcc/xcc_redfish_command.py

```diff
@@ -0,0 +1,599 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: xcc_redfish_command
+short_description: Manages Lenovo Out-Of-Band controllers using Redfish APIs
+version_added: 2.2.0
+description:
+  - Builds Redfish URIs locally and sends them to remote OOB controllers to
+    perform an action or get information back or update a configuration attribute.
+  - Manages virtual media.
+  - Supports getting information back via GET method.
+  - Supports updating a configuration attribute via PATCH method.
+  - Supports performing an action via POST method.
+options:
+  category:
+    required: true
+    description:
+      - Category to execute on OOB controller
+    type: str
+  command:
+    required: true
+    description:
+      - List of commands to execute on OOB controller
+    type: list
+  baseuri:
+    required: true
+    description:
+      - Base URI of OOB controller
+    type: str
+  username:
+    required: true
+    description:
+      - Username for authentication with OOB controller
+    type: str
+  password:
+    required: true
+    description:
+      - Password for authentication with OOB controller
+    type: str
+  timeout:
+    description:
+      - Timeout in seconds for URL requests to OOB controller
+    default: 10
+    type: int
+  resource_id:
+    required: false
+    description:
+      - The ID of the System, Manager or Chassis to modify
+    type: str
+  virtual_media:
+    required: false
+    description:
+      - The options for VirtualMedia commands
+    type: dict
+    suboptions:
+      media_types:
+        required: false
+        description:
+          - The list of media types appropriate for the image
+        type: list
+        elements: str
+      image_url:
+        required: false
+        description:
+          - The URL of the image to insert or eject
+        type: str
+      inserted:
+        required: false
+        description:
+          - Indicates if the image is treated as inserted on command completion
+        type: bool
+        default: True
```

Comment:
```suggestion
        default: true
```
Always use `true`/`false` for booleans in YAML.